src/TriggerActivityMakerTrackFinding.cpp

```diff
@@ -0,0 +1,303 @@
+    if (config.contains("print_tp_info"))
+      m_print_tp_info = config["print_tp_info"];
+    if (config.contains("prescale"))
+      m_prescale = config["prescale"]; 
+  }
+}
+
+TriggerActivity
+TriggerActivityMakerTrackFinding::construct_ta(TPWindow win_adj_max) const
+{
+  
+  TriggerActivity ta;
+  
+  TriggerPrimitive last_tp = win_adj_max.inputs.back();
+  
+  ta.time_start = last_tp.time_start;
+  ta.time_end = last_tp.time_start+last_tp.time_over_threshold;
+  ta.time_peak = last_tp.time_peak;
+  ta.time_activity = last_tp.time_peak;
+  ta.channel_start = last_tp.channel;
+  ta.channel_end = last_tp.channel;
+  ta.channel_peak = last_tp.channel;
+  ta.adc_integral = win_adj_max.adc_integral;
+  ta.adc_peak = last_tp.adc_integral;
+  ta.detid = last_tp.detid;
+  ta.type = TriggerActivity::Type::kTPC;
+  ta.algorithm = TriggerActivity::Algorithm::kTrackFinding;
+  ta.inputs = win_adj_max.inputs;
+
+  for( const auto& tp : ta.inputs ) {
+    ta.time_start = std::min(ta.time_start, tp.time_start);
+    ta.time_end = std::max(ta.time_end, tp.time_start+tp.time_over_threshold);
+    ta.channel_start = std::min(ta.channel_start, tp.channel);
+    ta.channel_end = std::max(ta.channel_end, tp.channel);
+    if (tp.adc_peak > ta.adc_peak ) {
+      ta.time_peak = tp.time_peak;
+      ta.adc_peak = tp.adc_peak;
+      ta.channel_peak = tp.channel;
+    }
+  }
+
+  return ta;
+}
+
+//std::vector<TriggerPrimitive>
+TPWindow
+TriggerActivityMakerTrackFinding::check_adjacency() {
+  // This function deals with tp window (m_current_window), select adjacent tps (with a channel gap from 0 to 4; sum of all gaps < m_adj_tolerance),
+  // checks if track length > m_adjacency_threshold: return the tp window (win_adj_max, which is subset of the input tp window)
+
+  unsigned int channel = 0;      // Current channel ID
+  unsigned int next_channel = 0; // Next channel ID
+  unsigned int next = 0;         // The next position in the hit channels vector
+  unsigned int tol_count = 0;    // Tolerance count, should not pass adj_tolerance
+
+  // Generate a channelID ordered list of hit channels for this window; second element of pair is tps
+  std::vector< std::pair <int, TriggerPrimitive> > chanTPList;
+  for (auto tp : m_current_window.inputs) {
+    chanTPList.push_back( std::make_pair(tp.channel, tp) );
+  }
+  std::sort(chanTPList.begin(), chanTPList.end(), [](const std::pair<int,TriggerPrimitive> &a, const std::pair<int,TriggerPrimitive> &b){return (a.first < b.first);});
+  
+  // ADAJACENCY LOGIC ====================================================================
+  // =====================================================================================
+  // Adjcancency Tolerance = Number of times prepared to skip missed hits before resetting 
+  // the adjacency count (win_adj). This accounts for things like dead channels / missed TPs. The 
+  // maximum gap is 4 which comes from tuning on December 2021 coldbox data, and June 2022 
+  // coldbox runs.
```

Comment:
Might be worth removing the maximum gap comment being tested on coldboxes. It applies to HMA, but not this algorithm

Reply:
done